Jenkinsfile

```diff
@@ -870,22 +870,28 @@ pass = ${OBS_CREDENTIALS_PASSWORD}
             }
             // Save logs of failed builds to s3 - we want to analyze where we may have jenkins issues.
             script {
-                if ((env.BRANCH_NAME == 'develop' || env.BRANCH_NAME == 'master') && fileExists('bin/clean-jenkins-log')) {
+                if (env.BRANCH_NAME == 'develop' || env.BRANCH_NAME == 'master') {
                     writeFile(file: 'build.log', text: getBuildLog())
                     sh 'bin/clean-jenkins-log'
+                    sh 'container-host-files/opt/scf/bin/klog.sh -f ${jobBaseName()}-${BUILD_NUMBER}-scf'
                     withAWS(region: params.S3_REGION) {
                         withCredentials([usernamePassword(
                             credentialsId: params.S3_CREDENTIALS,
                             usernameVariable: 'AWS_ACCESS_KEY_ID',
                             passwordVariable: 'AWS_SECRET_ACCESS_KEY',
                         )]) {
                             script {
-                                def subdir = "${params.S3_PREFIX}${distSubDir()}"
                                 def prefix = distPrefix()
+                                def subdir = "${params.S3_PREFIX}${distSubDir()}${prefix}${env.BUILD_TAG}/"
```

Comment:
Ok. @f0rmiga has convinced me that the current definition and path are not suitable.

The proposal is now this:

- Change "${params.S3_PREFIX}${distSubDir()}${prefix}${env.BUILD_TAG}/"
  to "${params.S3_PREFIX}${distSubDir()}${prefix}/${env.BUILD_TAG}/"
  This inserts `/` after the prefix.

- This has the problem that the path now looks like `PR-2170-/jenkins-scf-PR-2170-6`. Note the weird `-/` combination.

- To fix that remove the `-` from the output of the `distPrefix()` function. Should be around line 102 of Jenkinsfile.

- Now as `distPrefix()` is already used in other places, we have to compensate for that removal there, by adding the `-` back in where the output of `distPrefix()` is used. This is in the lines 816 (*), 821, 830. 

(*) In the line 816 it is the variable `prefix` holding the result from `distPrefix()` in line 809.

@bikramnehra ^

Reply:
@andreas-kupries Agreed, however, since all other cases require `-`. Therefore, instead of making changes in the `distPrefix()`(also so many other places), we can simply omit `-` while appending to the path:

`prefix = prefix.substring(0, prefix.length() - 1)`

just before line 885

Reply:
That is definitely less work to do. Go ahead. I see no future cases at the moment.

Reply:
DONE! Triggered a build to verify if everything's okay.